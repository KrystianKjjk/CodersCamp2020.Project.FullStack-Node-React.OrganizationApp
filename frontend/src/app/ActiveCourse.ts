import { CourseListElementModel } from "../components/CourseList/CourseListSlice";

export function setActiveCourse(course:CourseListElementModel){
localStorage.setItem("activeCourse", JSON.stringify(course));
}

export function getActiveCourse(): CourseListElementModel | null{
const activeCourseItem = localStorage.getItem("activeCourse");
if(!activeCourseItem){
    return null;
}
return JSON.parse(activeCourseItem);
}

