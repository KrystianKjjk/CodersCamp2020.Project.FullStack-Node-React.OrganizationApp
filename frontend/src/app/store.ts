import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import editingReducer from '../Redux/EditingSlice'
import reusableTableReducer from '../components/ReusableTable/ReusableTableSlice'
import homePageReducer from '../pages/Common/HomePage/HomePageSlice'
import courseDetails from '../pages/Admin/Course/CourseDetailsSlice'
import courseList from '../pages/Admin/CourseList/CourseListSlice'
import SnackbarReducer from '../components/Snackbar/SnackbarSlice'

export const store = configureStore({
  reducer: {
    editingInfo: editingReducer,
    tables: reusableTableReducer,
    userData: homePageReducer,
    courseDetails,
    courseList,
    snackbarDetails: SnackbarReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
