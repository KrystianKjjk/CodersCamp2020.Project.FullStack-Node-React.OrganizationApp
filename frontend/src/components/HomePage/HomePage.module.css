@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@700&display=swap');
.container{
    width:100%;
    text-align: left;
}
/* .container>div{
    padding-left: 3%;
    padding-right: 3%;

} */
.header{
    border: 1px solid #666666;
    padding: 1.5%;
    margin-bottom: 2%;
    background-color: #1C1C1C;
    padding-left: 3%;
    padding-right: 3%;
}

.description{
    border: 1px solid #666666;
    background-color: #1C1C1C;
    font-family: 'Montserrat', sans-serif;
    
}
.description div:first-child{
    border-bottom: 1px solid #666666;
    padding-left: 3%;
    padding-right: 3%;
    display: flex;
    align-items: center;
}
description div:first-child h2{
    padding-left:2%;
}
.description div:last-child{
   
    padding: 3% 3%;
}