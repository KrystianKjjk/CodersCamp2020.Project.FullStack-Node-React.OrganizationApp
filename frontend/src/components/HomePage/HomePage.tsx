import React from "react";
import styles from "./HomePage.module.css";
import NotificationsIcon from '@material-ui/icons/Notifications';

export interface HomePageProps {}

const HomePage: React.FC<HomePageProps> = (props) => {
  return (
    <div className={styles.container}>
      <div className={styles.header}>Hello Name !</div>
      <div className={styles.description}> 
        <div><NotificationsIcon fontSize="large" style={{paddingRight:"20px"}}></NotificationsIcon><h2> Next event</h2></div>
        <div><h2>Java Script Section Test - 22.01.2021</h2></div>
      </div>
    </div>
  );
};

export default HomePage;
