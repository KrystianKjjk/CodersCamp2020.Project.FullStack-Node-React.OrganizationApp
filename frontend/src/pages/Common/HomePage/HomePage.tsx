import React, {useEffect, useState} from "react";
import styles from "./HomePage.module.css";
import NotificationsIcon from '@material-ui/icons/Notifications';
import PageHeader from "../../../components/PageHeader";

export interface HomePageProps {}

const HomePage: React.FC<HomePageProps> = (props) => {

    const [course, setCourse] = useState<any>(null);

    useEffect(() => {
        setCourse(JSON.parse( localStorage.getItem('activeCourse')! ));
    }, []);

    return (
    <div className={styles.container}>
      <PageHeader name="Homepage"/>
      <div className={styles.description}> 
        <div><NotificationsIcon fontSize="large" style={{paddingRight:"20px"}}></NotificationsIcon><h2> Current event</h2></div>
        <div>
            {
                course ? (
                    <div style={{display: "block", textAlign: "center"}}>
                        <h1>{course?.name}</h1>
                        <h3 style={{marginTop: "2rem"}}>Start date: { new Date(course?.startDate).toLocaleDateString() }</h3>
                        <p style={{marginTop: "2rem"}}>{course?.description}</p>
                    </div>
                ) : (
                    <h2>Information soon!</h2>
                )
            }
        </div>
      </div>
    </div>
  );
};

export default HomePage;
