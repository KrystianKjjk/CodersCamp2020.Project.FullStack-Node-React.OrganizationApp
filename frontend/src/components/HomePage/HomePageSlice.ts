import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppThunk, RootState } from '../../app/store';

interface HomePageState {
  
}

const initialState: HomePageState = {
  
};

export const homePageSlice = createSlice({
  name: 'homePage',
  initialState,
  reducers: {

  },
});

export const { } = homePageSlice.actions;

// if you want, add selectors here, change the one below, remember to register reducer in store.ts
// export const selectHomePage = (state: RootState) => state.homePage.value;

export default homePageSlice.reducer;
