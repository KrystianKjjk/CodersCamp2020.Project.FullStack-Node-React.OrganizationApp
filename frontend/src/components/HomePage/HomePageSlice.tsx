import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

import UserService from "../../api/users.service";
import {RootState} from "../../app/store";

const usersService = new UserService();

const initialState: any = {
    loading: false,
    loaded: false,
    error: false,
    userData: {},
};

export const fetchUser: any = createAsyncThunk('fetchUserData', async (userID: string) => {
    const response = await usersService.getUserMe(userID)
    const userData = response.data;
    return userData;
});


export const homePageSlice = createSlice({
        name: 'homePage',
        initialState,
        reducers: {

        },
        extraReducers: {
            [fetchUser.pending]: (state) => {
                if (!state.loading) state.loading = true;
            },
            [fetchUser.fulfilled]: (state, action) => {
                if(state.loading) state.loading = false;
                state.loaded = true;
                state.userData = {...action.payload};
            },
            [fetchUser.rejected]: (state, action) => {
                if (state.loading) state.loading = false;
                state.error = action.error;
            },
        },
    }

);
export const selectUserData = (state: RootState) => state.userData;

export const {  } = homePageSlice.actions;
export default homePageSlice.reducer;
