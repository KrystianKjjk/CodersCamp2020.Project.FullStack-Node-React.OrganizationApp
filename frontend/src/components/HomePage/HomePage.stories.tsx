import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import HomePage , { HomePageProps } from './HomePage';

export default {
  title: 'HomePage component',
  component: HomePage,
} as Meta;

const Template: Story<HomePageProps> = (args) => <HomePage {...args} />;

export const FirstInput = Template.bind({});
FirstInput.args = {
  label: 'Write something: ',
};