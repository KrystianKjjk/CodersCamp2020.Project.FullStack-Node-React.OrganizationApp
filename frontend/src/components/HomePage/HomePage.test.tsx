import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import HomePage from './HomePage';

describe('HomePage', () => {
   it('renders without error', () => {
      render(
         <Provider store={store}>
            <HomePage />
         </Provider>
      );
   });
});