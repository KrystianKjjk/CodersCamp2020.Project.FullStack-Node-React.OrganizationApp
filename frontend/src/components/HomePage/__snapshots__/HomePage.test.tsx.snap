// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders without error 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      Hello Name !
    </div>
    <div
      class="description"
    >
      <div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          style="padding-right: 20px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
          />
        </svg>
        <h2>
           Next event
        </h2>
      </div>
      <div>
        <h2>
          Java Script Section Test - 22.01.2021
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage should render home page 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      Hello Name !
    </div>
    <div
      class="description"
    >
      <div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          style="padding-right: 20px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
          />
        </svg>
        <h2>
           Next event
        </h2>
      </div>
      <div>
        <h2>
          Java Script Section Test - 22.01.2021
        </h2>
      </div>
    </div>
  </div>
</div>
`;
